stock_picking_batch_oca/models/stock_batch_picking.py

```diff
@@ -0,0 +1,243 @@
+        default=lambda self: self.env.user.company_id.use_oca_batch_validation,
+        copy=False,
+    )
+
+    picking_ids = fields.One2many(
+        string='Pickings',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='List of picking managed by this batch.',
+    )
+    # TODO add comment to this field
+    active_picking_ids = fields.One2many(
+        string="Active Pickings",
+        comodel_name='stock.picking',
+        inverse_name='batch_id',
+        readonly=True,
+        domain=[('state', 'not in', ('cancel', 'done'))],
+    )
+
+    notes = fields.Text('Notes', help='free form remarks')
+
+    move_lines = fields.Many2many(
+        comodel_name='stock.move',
+        readonly=True,
+        string='Operations',
+        compute='_compute_move_lines',
+    )
+
+    move_line_ids = fields.Many2many(
+        comodel_name='stock.move.line',
+        string='Detailed operations',
+        compute='_compute_move_line_ids',
+        # HACK: Allow to write sml fields from this model
+        inverse=lambda self: self,
+    )
+
+    entire_package_ids = fields.Many2many(
+        comodel_name='stock.quant.package',
+        compute='_compute_entire_package_ids',
+        help='Those are the entire packages of a picking shown in the view of '
+             'operations',
+    )
+
+    entire_package_detail_ids = fields.Many2many(
+        comodel_name='stock.quant.package',
+        compute='_compute_entire_package_ids',
+        help='Those are the entire packages of a picking shown in the view of '
+             'detailed operations',
+    )
+
+    @api.depends('picking_ids')
+    def _compute_move_lines(self):
+        for batch in self:
+            batch.move_lines = batch.picking_ids.mapped("move_lines")
+
+    @api.depends('picking_ids')
+    def _compute_move_line_ids(self):
+        for batch in self:
+            batch.move_line_ids = batch.picking_ids.mapped(
+                'move_line_ids'
+            )
+
+    @api.depends('picking_ids')
+    def _compute_entire_package_ids(self):
+        for batch in self:
+            batch.update({
+                'entire_package_ids': batch.picking_ids.mapped(
+                    'entire_package_ids'),
+                'entire_package_detail_ids': batch.picking_ids.mapped(
+                    'entire_package_detail_ids'),
+            })
+
+    def get_not_empties(self):
+        """ Return all batches in this recordset
+        for which picking_ids is not empty.
+
+        :raise UserError: If all batches are empty.
+        """
+        if not self.mapped('picking_ids'):
+            if len(self) == 1:
+                message = _('This Batch has no pickings')
+            else:
+                message = _('These Batches have no pickings')
+
+            raise UserError(message)
+
+        return self.filtered(lambda b: len(b.picking_ids) != 0)
+
+    def verify_state(self, expected_state=None):
```

Comment:
As this method is called from standard picking methods (action_cancel, action_done...) it should not update related batches if `use_oca_batch_validation` is not set.

Reply:
verify_state called from action_cancel and odoo cancel method called cancel_picking, methods from odoo and this addon just different they are not inherited, they accessible only via form view which hide buttons in dependency of use_oca_batch_validation